pkgs/applications/window-managers/compton/git.nix

```diff
@@ -0,0 +1,55 @@
+{ stdenv, fetchFromGitHub, asciidoc, dbus, docbook_xml_dtd_45,
+  docbook_xml_xslt, libconfig, libdrm, libxml2, libxslt, mesa, pcre,
+  pkgconfig, libXcomposite, libXdamage, libXext, libXfixes, libXinerama,
+  libXrandr, libXrender }:
+
+stdenv.mkDerivation {
+  name = "compton-git-2015-04-20";
+
+  src = fetchFromGitHub {
+    owner  = "chjj";
+    repo   = "compton";
+    rev    = "b1889c1245e6f47eedfae6063100d5a16f584e2b";
+    sha256 = "0brnbidxi7wg08yiwgnijzcyqv5lnkd74xzfymvb0i7pgy465vaf";
+  };
+
+  buildInputs = [
+    asciidoc
+    dbus
+    docbook_xml_dtd_45
+    docbook_xml_xslt
+    libXcomposite
+    libXdamage
+    libXext
+    libXfixes
+    libXinerama
+    libXrandr
+    libXrender
+    libconfig
+    libdrm
+    libxml2
+    libxslt
+    mesa
+    pcre
+    pkgconfig
+  ];
+
+  buildFlagsArray = ["CFLAGS=-O3 -fomit-frame-pointer"];
```

Comment:
Bash arrays cannot be set from inside of Nix. If you really want an array, then the way to do it is to define

    preConfigure = ''
      buildFlagsArray=( ... )
    '';

I see no reason why an ordinary

    buildFlags = "CFLAGS=-O3 -fomit-frame-pointer";

definition shouldn't suffice, though.